plugins/by-name/copilot-lua/default.nix

```diff
@@ -1,172 +1,224 @@
 {
-  lib,
-  helpers,
   config,
+  lib,
   pkgs,
   ...
 }:
-with lib;
 let
-  cfg = config.plugins.copilot-lua;
-in
-{
-  options = {
-    plugins.copilot-lua =
-      let
-        keymapOption = helpers.defaultNullOpts.mkNullable (with types; either (enum [ false ]) str);
-      in
-      lib.nixvim.plugins.neovim.extraOptionsOptions
-      // {
-        enable = mkEnableOption "copilot.lua";
-
-        package = lib.mkPackageOption pkgs "copilot.lua" {
-          default = [
-            "vimPlugins"
-            "copilot-lua"
-          ];
-        };
-
-        panel = {
-          enabled = helpers.defaultNullOpts.mkBool true "Enable the panel.";
-
-          autoRefresh = helpers.defaultNullOpts.mkBool false "Enable auto-refresh.";
-
-          keymap = {
-            jumpPrev = keymapOption "[[" "Keymap for jumping to the previous suggestion.";
-
-            jumpNext = keymapOption "]]" "Keymap for jumping to the next suggestion.";
-
-            accept = keymapOption "<CR>" "Keymap to accept the proposed suggestion.";
+  inherit (lib) types;
+  inherit (lib.nixvim) defaultNullOpts;
 
-            refresh = keymapOption "gr" "Keymap to refresh the suggestions.";
-
-            open = keymapOption "<M-CR>" "Keymap to open.";
-          };
+  mkKeymapOption = defaultNullOpts.mkNullableWithRaw (with types; either (enum [ false ]) str);
```

Comment:
Would have probably preferred this in a let block for the `settingsOptions`

Reply:
Done.